syntax = "proto3";
package instance_relation;




import "cmdb_sdk/model/cmdb/auto_discovery_instance.proto";
import "google/protobuf/struct.proto";

/*
Discovery请求
*/
message DiscoveryRequest {
message Match {
    /*
    左侧实例匹配字段列表
    */
    repeated string left_match = 1 ;
    /*
    右侧实例匹配字段列表
    */
    repeated string right_match = 2 ;
}
        
message Data {
    /*
    左侧匹配的实例数据
    */
    google.protobuf.Struct left_instance = 1 ;
    /*
    右侧匹配的实例数据
    */
    google.protobuf.Struct right_instance = 2 ;
}
        
    
    /*
    关系Id
    */
    string relationId = 1 ;
    /*
    发现实例的字段列表
    */
    repeated Match match = 2 ;
    /*
    发现实例的数据列表
    */
    repeated Data data = 3 ;
    /*
    精确匹配（true: 只允许1:1关系，false: 允许1:n关系，默认true）
    */
    bool strict = 4 ;
}

/*
Discovery返回
*/
message DiscoveryResponse {
    /*
    状态码（0：成功，非0：失败）
    */
    int32 code = 1 ;
    /*
    [废弃字段] 关联的实例关系id
    */
    repeated string match_relation_instance_ids = 2 ;
    /*
    左侧实例详情
    */
    cmdb.AutoDiscoveryInstance left_instance = 3 ;
    /*
    右侧实例详情
    */
    cmdb.AutoDiscoveryInstance right_instance = 4 ;
}

/*
DiscoveryApi返回
*/
message DiscoveryResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    repeated instance_relation.DiscoveryResponse data = 4 ;
}

